stack/eval.go

```diff
@@ -31,11 +31,15 @@ type EvalCtx struct {
 }
 
 // NewEvalCtx creates a new stack evaluation context.
-func NewEvalCtx(rootdir string, sm Metadata, globals Globals) *EvalCtx {
-	evalctx := &EvalCtx{evalctx: eval.NewContext(sm.HostPath())}
-	evalctx.SetMetadata(rootdir, sm)
-	evalctx.SetGlobals(globals)
-	return evalctx
+func NewEvalCtx(rootdir string, sm Metadata, globals Globals) (*EvalCtx, error) {
+	evalctx, err := eval.NewContext(sm.HostPath())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Here since we are guaranteeing that the NewContext is always created with the stack host path which must always be valid maybe we could panic and keep the API as it was, seems like an internal/severe programming error if this failed.